Engine/Results/BaseResultsHandler.cs

```diff
@@ -667,6 +672,23 @@ protected virtual void SampleCapacity(DateTime time)
             return runtimeStatistics;
         }
 
+        /// <summary>
+        /// Gets the algorithm state data
+        /// </summary>
+        protected Dictionary<string, string> GetAlgorithmState(string endTime = "")
+        {
+            var state = new Dictionary<string, string>
+            {
+                ["Status"] = Algorithm.Status.ToStringInvariant(),
+                ["StartTime"] = StartTime.ToStringInvariant(),
+                ["Hostname"] = _hostName,
+                ["EndTime"] = endTime,
+                ["RuntimeError"] = Algorithm?.RunTimeError?.Message ?? "",
```

Comment:
Here it's doing a null check for the algorithm `Algorithm?` but in the other cases not, should probably null check it all of them?
Why the inner exception bellow?